File: OC/GifBackgroundViewDemo/GifBackgroundViewDemo/ViewController.m
//
//  ViewController.m
//  GifBackgroundViewDemo
//
//  Created by Vincent on 4/25/17.
//  Copyright © 2017 zssr. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIWebView *webView;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self loadHTML];
}

- (void)loadHTML {
    NSString *path = [[NSBundle mainBundle] pathForResource:@"background" ofType:@"html"];
    NSURL *url = [NSURL fileURLWithPath:path];
    NSData *data = [NSData dataWithContentsOfURL:url];
    NSURL *baseURL = [url URLByDeletingLastPathComponent];
    [self.webView loadData:data MIMEType:@"text/html" textEncodingName:@"utf-8" baseURL:baseURL];
}

- (UIStatusBarStyle)preferredStatusBarStyle {
    return UIStatusBarStyleLightContent;
}
@end

File: Swift/GifBackgroundViewDemo/GifBackgroundViewDemo/ViewController.swift
//
//  ViewController.swift
//  GifBackgroundViewDemo
//
//  Created by Vincent on 4/25/17.
//  Copyright © 2017 zssr. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    // override
    override var preferredStatusBarStyle: UIStatusBarStyle {
        return .lightContent
    }
    override func viewDidLoad() {
        super.viewDidLoad()
        loadHTML()
    }
    
    private func loadHTML() {
        let path = Bundle.main.path(forResource: "background", ofType: "html")
        let url = URL(fileURLWithPath: path!)
        let data = try? Data(contentsOf: url)
        print(path!)
        webView.load(data!, mimeType: "text/html", textEncodingName: "utf-8", baseURL: url.deletingLastPathComponent())
    }
    
    @IBOutlet weak var webView: UIWebView!
}

